lib/dal/src/standard_connection.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{
+    EdgeWeightError, EdgeWeightKind, EdgeWeightKindDiscriminants, TransactionsError,
+    WorkspaceSnapshotError,
+};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum HelperError {
+    #[error("invalid edge weight, got {0:?} expected discriminant {1:?}")]
+    InvalidEdgeWeight(EdgeWeightKind, EdgeWeightKindDiscriminants),
+    #[error("edge weight error: {0}")]
+    EdgeWeight(#[from] EdgeWeightError),
+    #[error("transactions error: {0}")]
+    Transactions(#[from] TransactionsError),
+    #[error("workspace snapshot error: {0}")]
+    WorkspaceSnapshot(#[from] WorkspaceSnapshotError),
+}
+
+pub type HelperResult<T> = Result<T, HelperError>;
```

Comment:
This is a nitpick, but I wonder if we want to name this `AddEdgeError` or something similar? I'm wondering if it would help with traces and logging.